app/src/main/java/net/sourceforge/opencamera/ui/MainUI.java

```diff
@@ -2451,23 +2457,9 @@ public void toggleRecSyncSettings() {
         if( rec_sync_view == null ) {
             if( MyDebug.LOG )
                 Log.d(TAG, "create new rec_sync_view");
-            rec_sync_view = new RecSyncView(main_activity);
-
-            rec_sync_view.setSyncSettingsOnClickListener(view -> {
-                if (MyDebug.LOG)
-                    Log.d(TAG, "clicked to buttonSyncSettings");
-
-                main_activity.clickedSyncSettings();
-            });
-
-            rec_sync_view.setAlignPhasesOnClickListener(view -> {
-                if (MyDebug.LOG)
-                    Log.d(TAG, "clicked to buttonAlignPhases");
-
-                main_activity.clickedAlignPhases();
-            });
 
-            rec_sync_container.addView(rec_sync_view);
+            LayoutInflater inflater = main_activity.getLayoutInflater();
+            rec_sync_view = inflater.inflate(R.layout.recsyncview_layout, rec_sync_container);
```

Comment:
I think that it isn't a great solution - I mentioned that we should use inflating, but it doesn't mean that the custom widget should be removed! Just inflate XML in init block in your widget and use it as before (only without UI construction methods in constructor)

Reply:
Then I have some questions here:
1) Is that a bad solution because it is less flexible (not having a special class for the view)?
2) If we don't create a view by inflating it as a whole as I did here, then, as far as I understand, we can only inflate buttons and add them to the view manually. So the only difference will be that we don't have to setup buttons. Is that what you mean?

Reply:
1) Yes, because all logic of click listeners and information about UI elements inside view (like buttons) will be incapsulated
2) Yes, if you inflate only buttons you will only optimize their setup. I mean here, that possible solution now is to inflate the whole view but inside its class. So your view will be extending frame layout and you inflate your XML with root attaching (second boolean argument in inflate method)